django/middleware/common.py

```diff
@@ -74,11 +79,12 @@ def should_redirect_with_slash(self, request):
         Return True if settings.APPEND_SLASH is True and appending a slash to
         the request path turns an invalid path into a valid one.
         """
-        if settings.APPEND_SLASH and not request.path_info.endswith('/'):
+        path_info = '' if request.path_info_is_empty else request.path_info
```

Comment:
rather than the new `path_info_is_empty` attribute, would `path_info = request.META.get('PATH_INFO', '')` work here?